app/services/migration/migrators/application_state.rb

```diff
@@ -0,0 +1,37 @@
+module Migration::Migrators
+  class ApplicationState < Base
+    class << self
+      def record_count
+        ecf_participant_profile_states.count
+      end
+
+      def model
+        :application_state
+      end
+
+      def dependencies
+        %i[application]
+      end
+
+      def ecf_participant_profile_states
+        Migration::Ecf::ParticipantProfileState
+          .joins(participant_profile: :teacher_profile)
+          .includes(:cpd_lead_provider)
+          .where(teacher_profile: { user_id: User.ecf_users.pluck(:id) })
+      end
+    end
+
+    def call
+      migrate(self.class.ecf_participant_profile_states) do |ecf_participant_profile_state|
+        ecf_application_id = ecf_participant_profile_state.participant_profile_id
+        application_id = ::Application.select(:id).find_by!(ecf_id: ecf_application_id).id
+        application_state = ::ApplicationState.find_or_initialize_by(application_id:)
```

Comment:
so I believe an application can have multiple states, so we might need to use the ecf id of the profile state to find or initialise by instead